### PR TITLE
Fix! : Adding test and refactor #40

### PR DESCRIPTION
**Pull request by bluedrack**
the code to achieves the 80 pour-cent code coverage. Adding packages to clean up the code base by adding for each activity with their corresponding Module and Interface for code injection with Hilt. Like discuss in the previous meeting.

### CODE REVIEW

File: app/src/main/java/com/github/studydistractor/sdp/register/RegisterAuthModule.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import dagger.hilt.android.components.ActivityComponent
+
+@Module
+@InstallIn(ActivityComponent::class)
+object RegisterAuthModule {
+    @Provides
+    fun provideRegisterAuth() : RegisterAuthInterface {
+        return FirebaseRegisterAuth()
+    }
+//    @Provides
```
Comment (lipefree) [inline app/src/main/java/com/github/studydistractor/sdp/register/RegisterAuthModule.kt:19]: Remove comment

**Review by lipefree — CHANGES_REQUESTED** [review]
Overall efficient code, the repackaging is nice ! But include some class and function documentation in stuff you wrote in this PR.

**Review by abouquet27 — COMMENTED** [review]
This commit is a mix of refactoring, adding tests and modifications in the register class. The description of the commit should be more explicit and must list precisely all the major modifications because we cannot understand what is going on, specially if we are working on those files too

**Comment by bluedrack** [comment]
> This commit is a mix of refactoring, adding tests and modifications in the register class. The description of the commit should be more explicit and must list precisely all the major modifications because we cannot understand what is going on, specially if we are working on those files too



> This commit is a mix of refactoring, adding tests and modifications in the register class. The description of the commit should be more explicit and must list precisely all the major modifications because we cannot understand what is going on, specially if we are working on those files too

Yes I agree with you but I need to do that to implement the hilt implementation so it is necessary to add Module and Interface so that the testing will be more easy later.

**Review by abouquet27 — CHANGES_REQUESTED** [review]
Great modularisation of the code and it is a better structural tree. The project is more clearer and the tests are good. 

Apart from the comment I posted earlier, it is fine for me
LGTM

**Review by lipefree — APPROVED** [review]
LGTM now !

**Pull request merged**